src/main/java/racingcar/service/RacingGameService.java

```diff
@@ -0,0 +1,49 @@
+package racingcar.service;
+
+
+import racingcar.domain.Cars;
+import racingcar.domain.GameResult;
+import racingcar.domain.RacingGame;
+import racingcar.dto.CarDto;
+import racingcar.dto.CarsDto;
+import racingcar.dto.WinnersDto;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.collectingAndThen;
+
+public class RacingGameService implements GameService {
+    private RacingGame racingGame;
+
+    @Override
+    public void setUpGame(List<String> userNames, int goalRound) {
+        racingGame = new RacingGame(userNames, goalRound);
+    }
+
+    @Override
+    public void setUpGame(Cars users, int goalRound) {
+        racingGame = new RacingGame(users, goalRound);
+    }
+
+    @Override
+    public CarsDto executeRound() {
+        racingGame.playRound();
+        Cars cars = racingGame.getCars();
+        return cars.getCars()
+                .stream()
+                .map(car -> new CarDto(car.getName().toString(), car.getPosition().index()))
+                .collect(collectingAndThen(Collectors.toList(), carDtos -> new CarsDto(carDtos, racingGame.getCurrentRound().getRound())));
```

Comment:
```suggestion
                .map(car -> new CarDto(car.getName(), car.getPositionIndex()))
                .collect(collectingAndThen(Collectors.toList(), carDtos -> new CarsDto(carDtos, racingGame.getCurrentRound())));
```

위와 같이 동작하도록 수정하면 어떠한 장단점이 있을지 고민해보아요

Reply:
제가 요즘 헤드퍼스트 디자인패턴을 읽고있는데, 거기서 최소지식원칙에 대해 읽었습니다. 뎁스가 늘어날수록 해당 클래스가 알아야 하는 정보도 많아지고, 이는 나중에 유지보수 소요를 가져올거 같네요!

Reply:
맞습니다!
매개변수로 넘겨줄 때도 해당하지만, 메서드의 반환값도 해당이 되는데요!
이러한 관점으로 car.getName()에서 원시값을 응답하면 어떨까요!?

Reply:
`매개변수로 넘겨줄 때도 해당하지만, 메서드의 반환값도 해당이 되는데요!`
무슨 관계지........... 고민하다가 아! 받는 쪽에서 Name에 대해서도 알아야 되는구나, 깨우쳤습니다.
VO를 활용하더라도 도메인 객체 내부에서 활용하면 되지, 밖에다가 칠렐레 팔렐레 뿌리고 다닐 필요는 없겠네요.